lib/sshkit/context.ex

```diff
@@ -22,32 +22,27 @@ defmodule SSHKit.Context do
   ```
   """
   def build(context, command) do
-    command
-    |> cmd
-    |> group(context.group)
-    |> user(context.user)
-    |> env(context.env)
+    "/usr/bin/env #{command}"
+    |> sudo(context.user, context.group)
+    |> export(context.env)
     |> umask(context.umask)
-    |> path(context.path)
+    |> cd(context.path)
   end
 
-  defp cmd(command), do: "/usr/bin/env #{command}"
+  defp sudo(command, nil, nil), do: command
+  defp sudo(command, username, nil), do: "sudo -n -u #{username} -- sh -c #{shellquote(command)}"
+  defp sudo(command, nil, groupname), do: "sudo -n -g #{groupname} -- sh -c #{shellquote(command)}"
+  defp sudo(command, username, groupname), do: "sudo -n -u #{username} -g #{groupname} -- sh -c #{shellquote(command)}"
```

Comment:
ah, you're right. i just misread the code yesterday evening. let's act as if i never wrote this comment ;)